metrics/ci_integrations/test/source/buildkite/config/validator/buildkite_source_validator_test.dart

```diff
@@ -0,0 +1,556 @@
+    }
+
+    PostExpectation<Future<InteractionResult>> whenValidateOrganizationSlug({
+      BuildkiteToken withAccessToken,
+    }) {
+      whenValidateAuth().thenSuccessWith(withAccessToken);
+
+      return when(
+        validationDelegate.validateOrganizationSlug(organizationSlug),
+      );
+    }
+
+    PostExpectation<Future<InteractionResult>> whenValidatePipelineSlug() {
+      whenValidateOrganizationSlug(
+        withAccessToken: allRequiredScopesToken,
+      ).thenSuccessWith(null);
+
+      return when(
+        validationDelegate.validateSourceProjectId(pipelineSlug),
+      );
+    }
+
+    /// Returns a [FieldValidationResult] for the
+    /// [BuildkiteSourceConfigField.accessToken] from the given [result].
+    FieldValidationResult _getAuthValidationResult(ValidationResult result) {
+      return result.results[BuildkiteSourceConfigField.accessToken];
+    }
+
+    /// Returns a [FieldValidationResult] for the
+    /// [BuildkiteSourceConfigField.pipelineSlug] from the given [result].
+    FieldValidationResult _getPipelineSlugValidationResult(
+      ValidationResult result,
+    ) {
+      return result.results[BuildkiteSourceConfigField.pipelineSlug];
+    }
+
+    /// Returns a [FieldValidationResult] for the
+    /// [BuildkiteSourceConfigField.organizationSlug] from the given [result].
+    FieldValidationResult _getOrganizationSlugValidationResult(
+      ValidationResult result,
+    ) {
+      return result.results[BuildkiteSourceConfigField.organizationSlug];
+    }
+
+    test(
+      "throws an ArgumentError if the given validation delegate is null",
+      () {
+        expect(
+          () => BuildkiteSourceValidator(null, validationResultBuilder),
+          throwsArgumentError,
+        );
+      },
+    );
+
+    test(
+      "throws an ArgumentError if the given validation result builder is null",
+      () {
+        expect(
+          () => BuildkiteSourceValidator(validationDelegate, null),
+          throwsArgumentError,
+        );
+      },
+    );
+
+    test(
+      "creates a new instance with the given parameters",
+      () {
+        final validator = BuildkiteSourceValidator(
+          validationDelegate,
+          validationResultBuilder,
+        );
+
+        expect(validator.validationDelegate, equals(validationDelegate));
+        expect(
+          validator.validationResultBuilder,
+          equals(validationResultBuilder),
+        );
+      },
+    );
+
+    test(
+      ".validate() delegates the access token validation to the validation delegate",
+      () {
+        whenValidateAuth().thenSuccessWith(noOrganizationScopeToken);
+
+        final expectedAuth = BearerAuthorization(accessToken);
+
+        validator.validate(config);
+
+        verify(validationDelegate.validateAuth(expectedAuth)).called(1);
+      },
+    );
+
+    test(
+      ".validate() returns a validation result with the successful access token field validation result if the access token is valid",
```

Comment:
Would it be better to test that the validate method sets the right auth field validation result after auth validation?